sdk/dotnet/Pulumi.Automation/PulumiFn.Inline.cs

```diff
@@ -16,6 +16,7 @@ internal sealed class PulumiFnInline : PulumiFn
         public PulumiFnInline(Func<CancellationToken, Task<IDictionary<string, object?>>> program)
         {
             this._program = program;
+            this.DiscoveryAssembly = Assembly.GetEntryAssembly();
```

Comment:
I waffled a bit on what the delegate-driven inline program should use as the default assembly, if anything. It can be hard to nail down a single assembly since delegates can be chained and added to from different assemblies and we don't know where it was started. I wanted to do `GetCallingAssembly()` in the `PulumiFn.Create(...)` factory methods but was a bit nervous about compiler optimization making that unpredictable. This is at least predictable, so do we think it is a reasonable default?